fem/coefficient.cpp

```diff
@@ -28,6 +28,52 @@ double PWConstCoefficient::Eval(ElementTransformation & T,
    return (constants(att-1));
 }
 
+void PWCoefficient::InitMap(const Array<int> & attr,
+                            const Array<Coefficient*> & coefs)
```

Comment:
So this is more of a question than anything, but I thought `Array<T>` had at some point moved over to POD, and I guess looking at the documentation it appears to now just require `std::is_trivial` for type T. However, I guess my question is there a preference for using `Array` here over a `std::vector` especially since it's not like the classes being used here can even be used on the GPU and it's only used for initialization / updating of the class?

Reply:
Just as another thought, it almost seems like you could eliminate the size checks between the two by just having the above be a `std::map<int, Coefficient*>` or `std::unordered_map<int, Coefficient*>`. Although, I guess there seems to be an aversion to the use of `std` containers in MFEM functions which might be why that's not being done here?

Reply:
Thanks for the feedback, @rcarson3. An `mfem::Array<int>` seemed the natural choice for the `attr` argument since this is how we store these integers in the mesh class. We have flexibility with the `coefs` argument, we could have used an `std::vector` or even a c-style array of pointers, but I felt it would feel more predictable to use the same container for both arguments.

I do like using `std::map` objects and this might be a useful alternative. I'm torn. I suspect users won't have pre-existing containers filled with the attribute/coefficient pairs that this class needs so they will need to create this before calling (or just use the other function to add them one-by-one). In this case it doesn't matter much which container we ask them to populate. I guess it comes down to what will feel most natural in MFEM and, for good or ill, this is probably `mfem::Arrays`.

Reply:
@mlstowell thanks for the explanation. I'll go ahead and approve this, and also who knows maybe in a future major release of MFEM some of these sorts of things can be revisited especially if the we see versions of the std library usable on more GPU systems than just nvidia's...